mlir/unittests/Dialect/MIOpen/CoordTransformBuilderTests.cpp

```diff
@@ -0,0 +1,220 @@
+//===- CoordTransformBuilderTests.cpp - Tests for the MIOpen Coordinate
+// Transform Builder -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/MIOpen/MIOpen.h"
+#include "mlir/IR/MLIRContext.h"
+#include "llvm/ADT/SmallVector.h"
+
+#include "gtest/gtest.h"
+#include "gtest/internal/gtest-internal.h"
+
+using namespace mlir;
+using namespace mlir::miopen;
+
+//===----------------------------------------------------------------------===//
+// Test Fixture
+//===----------------------------------------------------------------------===//
+
+class CTBuilderTest : public ::testing::Test {
```

Comment:
It's nice to see addition in `unittests` directory. Could you share some experiences / rationales between `test` and `unittests`?

Reply:
Well, the idea of the unit tests was to let me directly test functionality that wasn't necessarily exposed in the input/output to a pass, namely the coordinate transform builders themselves and how they generate affine maps. (Sure, we can check for a given convolution that we get the correct `transform_map`, but  I wanted to test the builders themselves before I started refactoring lowering).